src/main/java/org/folio/services/invoice/InvoiceCancelService.java

```diff
@@ -224,4 +232,106 @@ private Future<Void> unreleaseEncumbrancesForPoLines(List<PoLine> poLines, Invoi
       });
   }
 
+  private Future<Void> updatePoLinePaymentStatus(Invoice invoiceFromStorage, List<InvoiceLine> invoiceLines,
+      RequestContext requestContext) {
+    if (!Invoice.Status.PAID.equals(invoiceFromStorage.getStatus()) || invoiceFromStorage.getFiscalYearId() == null) {
+      // in the unlikely case the fiscal year is undefined, the payment status update is skipped
+      // (the MODINVOSTO-177 script should fill it up for pre-Poppy invoices)
+      return succeededFuture();
+    }
+    List<String> allPoLineIds = invoiceLines.stream()
+      .map(InvoiceLine::getPoLineId)
+      .filter(Objects::nonNull)
+      .distinct()
+      .toList();
+    if (allPoLineIds.isEmpty()) {
+      return succeededFuture();
+    }
+    return getPoLinesByIdAndQuery(allPoLineIds, this::queryToGetPoLinesWithFullyOrPartiallyPaidPaymentStatusByIds, requestContext)
+      .compose(filteredPoLines -> {
+        if (filteredPoLines.isEmpty()) {
+          return succeededFuture();
+        }
+        List<String> filteredPoLineIds = filteredPoLines.stream().map(PoLine::getId).toList();
+        return getInvoiceLinesByPoLineIdsAndQuery(filteredPoLineIds,
+            ids -> queryToGetRelatedPaidInvoiceLinesByPoLineIds(invoiceFromStorage.getId(), ids), requestContext)
+          .compose(relatedInvoiceLines -> {
+            if (relatedInvoiceLines.isEmpty()) {
```

Comment:
a bit complex logic, can you please add info logs into conditional statements that can help in troubleshooting

Reply:
Added more logging.